prime-router/src/main/kotlin/azure/SubmissionFunction.kt

```diff
@@ -82,68 +86,89 @@ class SubmissionFunction(
      */
-    @FunctionName("getOrgSubmissions")
-    fun getOrgSubmissions(
+    @FunctionName("getOrgSubmissionsList")
+    fun getOrgSubmissionsList(
         @HttpTrigger(
-            name = "getOrgSubmissions",
+            name = "getOrgSubmissionsList",
             methods = [HttpMethod.GET],
             authLevel = AuthorizationLevel.ANONYMOUS,
             route = "waters/org/{organization}/submissions"
         ) request: HttpRequestMessage<String?>,
         @BindingName("organization") organization: String,
     ): HttpResponseMessage {
-        return oktaAuthentication.checkAccess(request, organization, true) {
-            try {
-                val (qSortOrder, qSortColumn, resultsAfterDate, resultsBeforeDate, pageSize, showFailed) =
-                    Parameters(request.queryParameters)
-                val sortOrder = try {
-                    SubmissionAccess.SortOrder.valueOf(qSortOrder)
-                } catch (e: IllegalArgumentException) {
-                    SubmissionAccess.SortOrder.DESC
-                }
+        try {
+            // Do authentication
+            val claims = AuthenticationStrategy.authenticate(request)
+                ?: return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
+
+            // Confirm the org name in the path is a sender in the system.
+            val sender = workflowEngine.settings.findSender(organization) // err if no default sender in settings in org
+                ?: return HttpUtilities.unauthorizedResponse(
+                    request, HttpUtilities.errorJson("Authentication Failed: $organization: unknown sender")
+                )
 
-                val sortColumn = try {
-                    SubmissionAccess.SortColumn.valueOf(qSortColumn)
-                } catch (e: IllegalArgumentException) {
-                    SubmissionAccess.SortColumn.CREATED_AT
-                }
-                val submissions = submissionsFacade.findSubmissionsAsJson(
-                    organization,
-                    sortOrder,
-                    sortColumn,
-                    resultsAfterDate,
-                    resultsBeforeDate,
-                    pageSize,
-                    showFailed
+            // Do authorization based on: org name in the path == org name in claim.  Or be a prime admin.
+            if ((claims.organizationNameClaim != sender.organizationName) && !claims.isPrimeAdmin) {
+                logger.warn(
+                    "Invalid Authorization for user ${claims.userName}:" +
+                        " ${request.httpMethod}:${request.uri.path}." +
+                        " ERR: Claim org is ${claims.organizationNameClaim} but client id is ${sender.organizationName}"
                 )
-                HttpUtilities.okResponse(request, submissions)
+                return HttpUtilities.unauthorizedResponse(request, authorizationFailure)
+            }
+            logger.info(
```

Comment:
Change to debug or trace

Reply:
Per other comments, keeping this as is, to ensure log info is available in case of a security issue.